electives/18344.md

```diff
@@ -0,0 +1,45 @@
+# 18-344: Computer Systems and the Hardware/Software Interface
+
+| Category | Difficulty |
+|:-:       | :-:        |
+| Labs     | 6          |
+| Midterm  | 6          |
+
+This is a class that takes pieces of 18-240 and 18-213 and extends them, giving a higher-level view of computer architecture as a whole. It is a great option to take before 447 since there are many topics (CPU pipeline, branch prediction, memory hierarchy, parallelism) which 344 covers from a "why are they interesting and how do they affect software" perspective, and then 447 subsequently goes into depth on how they are actually implemented.
+
+The course starts where the last part of 240 leaves off - building up from a single-cycle CPU into a 5-stage pipelined RISC-V CPU with a memory hierarchy and a branch predictor. Unlike 240/341/447 however, you won't be writing any SystemVerilog code in this course - all the labs are done by writing code to simulate and analyze different components of a computer architecture. The later parts of this course focus more on the software side - all the topics which 213 can't quite go into depth on: more-advanced caches, virtual memory, different techniques for parallelism, and how to optimize different types of code.
```

Comment:
Or you can use a real emdash a well `—`, otherwise I like `--` or `---` in my code

```suggestion
The course starts where the last part of 240 leaves off - building up from a single-cycle CPU into a 5-stage pipelined RISC-V CPU with a memory hierarchy and a branch predictor. Unlike 240/341/447 however, you won't be writing any SystemVerilog code in this course -- all the labs are done by writing code to simulate and analyze different components of a computer architecture. The later parts of this course focus more on the software side -- all the topics which 213 can't quite go into depth on: more advanced caches, virtual memory, different techniques for parallelism, and how to optimize different types of code.
```

Reply:
fixed with c6c2e51